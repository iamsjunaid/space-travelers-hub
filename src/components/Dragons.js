import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { cancelReservation, reserveDragon } from '../redux/dragons/dragonsSlice';
import '../styles/Dragon.css';

function Dragons({
  id, name, type, image, reserved,
}) {
  const dispatch = useDispatch();

  return (
    <li className="dragon">
      <img src={image} alt={name} />
      <div>
        <p>{id}</p>

        <p>
          {reserved && <span className="reserved">Reserved</span>}
          {' '}
          <span className="heading">
            {name}
          </span>
        </p>
        <p>
          <span>
            {type}
          </span>
        </p>

        {!reserved && (
        <button className="reserve-btn" type="button" onClick={() => dispatch(reserveDragon(id))}>
          Reserve Dragon
        </button>
        )}
        {reserved && (
        <button className="cancel" type="button" onClick={() => dispatch(cancelReservation(id))}>
          Cancel Reservation
        </button>
        )}
      </div>
    </li>

  );
}

Dragons.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Dragons;
