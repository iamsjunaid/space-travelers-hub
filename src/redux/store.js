import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
    dragons: dragonsReducer,
  },
});

export default store;
