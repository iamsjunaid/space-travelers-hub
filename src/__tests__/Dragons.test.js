import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import Dragons from '../components/Dragons';
import store from '../redux/store';

test('renders Dragons component correctly', () => {
  const dragon = {
    id: '1',
    name: 'Dragon Name',
    type: 'Dragon Type',
    image: 'dragon.jpg',
    description: 'Dragon Description',
    reserved: false,
  };

  const { container } = render(
    <Provider store={store}>
      <Dragons dragons={dragon} />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
