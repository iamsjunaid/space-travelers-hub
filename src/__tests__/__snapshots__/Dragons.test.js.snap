// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dragons component correctly 1`] = `
<div>
  <li
    class="dragon"
  >
    <img />
    <div>
      <p>
         
        <span
          class="heading"
        />
      </p>
      <p>
        <span />
      </p>
      <p>
        <span />
      </p>
      <button
        class="reserve-btn"
        type="button"
      >
        Reserve Dragon
      </button>
    </div>
  </li>
</div>
`;
